packages/ciphernode/Cargo.toml

```diff
@@ -17,83 +17,66 @@ members = [
   "test_helpers",
   "tests",
 ]
+resolver = "2"
+
+[workspace.package]
+version = "0.1.0"
+edition = "2021"
 
 [workspace.dependencies]
 actix = "0.13.5"
 actix-web = "4.10.2"
 aes-gcm = "0.10.3"
-aggregator = { path = "./aggregator" }
-alloy = { version = "0.5.2", features = ["full", "node-bindings"] }
-alloy-primitives = { version = "0.6", default-features = false, features = [
-  "rlp",
-  "serde",
-  "std",
-] }
-alloy-sol-types = { version = "0.6" }
+alloy = "0.8"
+alloy-primitives = "0.8"
+alloy-sol-types = "0.6"
+alloy-dyn-abi = "1.1.0"
 argon2 = "0.5.2"
 anyhow = "1.0.86"
-async-std = { version = "1.12", features = ["attributes"] }
+async-std = "1.12"
 async-trait = "0.1"
 bincode = "1.3.3"
 bloom = "0.3.2"
 bs58 = "0.5.1"
 base64 = "0.22.1"
-clap = { version = "4.5.17", features = ["derive"] }
-crypto = { path = "./crypto" }
+clap = "4.5.17"
 compile-time = "0.2.0"
-config = { path = "./config" }
 dirs = "5.0.1"
-data = { path = "./data" }
 dialoguer = "0.11.0"
-events = { path = "./events" }
-enclave_core = { path = "./enclave_core" }
-evm = { path = "./evm" }
-fhe = { path = "./fhe" }
-fhe_rs = { package = "fhe", git = "https://github.com/gnosisguild/fhe.rs", version = "0.1.0-beta.7" }
-fhe-traits = { git = "https://github.com/gnosisguild/fhe.rs", version = "0.1.0-beta.7" }
-fhe-util = { git = "https://github.com/gnosisguild/fhe.rs", version = "0.1.0-beta.7" }
-figment = { version = "0.10.19", features = ["yaml", "test"] }
+fhe_rs = { package = "fhe", git = "https://github.com/gnosisguild/fhe.rs", branch = "feature/greco-integration" }
+fhe-traits = { git = "https://github.com/gnosisguild/fhe.rs", branch = "feature/greco-integration" }
+fhe-math = { git = "https://github.com/gnosisguild/fhe.rs.git", branch = "feature/greco-integration" }
+fhe-util = { git = "https://github.com/gnosisguild/fhe.rs", branch = "feature/greco-integration" }
+figment = "0.10.19"
 futures = "0.3.30"
 futures-util = "0.3"
 hex = "0.4.3"
-keyshare = { path = "./keyshare" }
 lazy_static = "1.5.0"
-logger = { path = "./logger" }
 num = "0.4.3"
-net = { path = "./net" }
+num-bigint = "0.4.6"
+num-traits = "0.2"
 once_cell = "1.19.0"
-opentelemetry = { version = "0.29.1" }
-opentelemetry-otlp = { version = "0.29.0", features = ["grpc-tonic"] }
```

Comment:
I believe we get faster compilation (untested) if we unify the feature set and there is not much downside to this in most situations despite considerations of occasionally having extra unused features.